internal/pkg/cli/deploy.go

```diff
@@ -208,6 +276,146 @@ func (o *deployOpts) askName() error {
 	return nil
 }
 
+func (o *deployOpts) askEnv() error {
+	if o.envName != "" {
+		return nil
+	}
+	localEnvs, err := o.ws.ListEnvironments()
```

Comment:
nit: what do you think of caching these results to reduce the number of disk reads?